### ISSUE TITLE
Support of creating custom neural network structure

### ISSUE THREAD

**Issue by gy2256** [pnnl/neuromancer#117]
Hello,

Thanks for creating this awesome library. Here is my question: 
I know there are already some customizations that can be applied when creating a NN. ([https://github.com/pnnl/neuromancer/blob/78a111858d892b8869827746acf1114928ab78bb/src/neuromancer/modules/blocks.py#L725-L738](https://github.com/pnnl/neuromancer/blob/78a111858d892b8869827746acf1114928ab78bb/src/neuromancer/modules/blocks.py#L725-L738))

Is there a way for me to create custom NN structures? For example, I want to bound the NN output using torch.clamp().

**Comment by drgona** [comment]
@gy2256  thank you for your interest in our work!

Neuromancer supports arbitrary level of customization of your computational graph. Anything from custom neural architectures implemented as nn.Module classes to custom python callable can be wrapped in our Node and Variable classes to make them part of the symbolic computational graph on which you can impose custom constraints and loss functions. 

Here are some examples of this customization:

1. Using lambda expression to implement linear state space model for learning to control task: https://github.com/pnnl/neuromancer/blob/master/examples/control/Part_1_stabilize_linear_system.ipynb
2. Gray-box differential equation implemented as a special instance of our ode.ODESystem class: https://github.com/pnnl/neuromancer/blob/master/examples/ODEs/Part_3_UDE.ipynb
3. Differentiable projection layer implemented as nn.Module class for learning to optimize task: https://github.com/pnnl/neuromancer/blob/master/examples/parametric_programming/Part_4_projectedGradient.ipynb 
4. Neural Lyapunov function architecture implemented as a combination of input-convex and positive-definite networks implemented as nn.Modules: https://github.com/pnnl/neuromancer/blob/master/examples/control/Part_5_neural_Lyapunov.ipynb

If you implement new neural architectures that you find helpful in your applications, we will welcome external contributions to the library. You can create new pull request with your new neural modules to be included in [blocks.py](https://github.com/pnnl/neuromancer/blob/master/src/neuromancer/modules/blocks.py)
